module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -2185,6 +2191,63 @@ public function changeEmailAction()
         return $view;
     }
 
+    /**
+     * Generate an API key for a user.
+     *
+     * @return mixed
+     */
+    public function generateApiKeyAction()
```

Comment:
Is it worth considering creating an ApiKeyController, just to prevent further growth of the already-huge MyResearchController, and to allow us to use a more meaningful template path?